pkg/engine/task/apply_conventions.go

```diff
@@ -139,3 +131,41 @@ func setControllerReference(owner v1.Object, obj runtime.Object, scheme *runtime
 	}
 	return nil
 }
+
+// render method takes resource names and Instance parameters and then renders passed templates using kudo engine.
+func render(resourceNames []string, templates map[string]string, params map[string]string, meta ExecutionMetadata) (map[string]string, error) {
```

Comment:
also the rendering, I would not have it inside apply_conventions. I would like to see a package for rendering (see my post in the kubernetes slack in kudo channel). Again does not have to be in this PR :)